### PR TITLE
[caffe2] uprade IDEEP and hotfix for conv op accuracy issue

### PR DESCRIPTION
**Pull request by gujinghui**
1. upgrade IDEEP to latest version
2. hotfix for the accuracy issue in IDEEP conv op

### CODE REVIEW

**Comment by gujinghui** [comment]
@yinghai 
Pls review

**Comment by gujinghui** [comment]
@jgong5

**Review by yinghai — COMMENTED** [review]
Could you please state the root cause of the issue and how the fix works? Thanks.

**Comment by gujinghui** [comment]
@yinghai 
1. the group filter in ideep is incomplete.
This causes the compute API is not reentant for same weight with group.
This issue has been fixed in latest ideep.

2. The assignment operator of tensor is a shadow copy.
tensor A = B   ----> A and B share same data buffer.
Then, the reorder operator does not support inplace case.

File: caffe2/ideep/operators/conv_op.cc [hunk 1]
```diff
@@ -40,15 +40,20 @@ class IDEEPConvOp final : public IDEEPConvPoolOpBase {
         (cached_weights_descriptor_ != filter.get_descriptor());
     if (weights_changed && !training_mode_) {
       cached_weights_descriptor_ = filter.get_descriptor();
-      filter_ = filter;
+      auto filter_in = filter;
+      filter_in.make_group(group_);
```
Comment (yinghai) [inline caffe2/ideep/operators/conv_op.cc]: Why do we need this line? Isn't the shape of filter_ determined by the next line?

Comment (yinghai) [reply caffe2/ideep/operators/conv_op.cc]: Never mind. Looks like it's changing the descriptor underlying, which means it's needed:
https://github.com/intel/ideep/blob/4bd9a6800bf7db068187619e0582d34dec9651dc/include/ideep/tensor.hpp#L835

**Comment by yinghai** [comment]
@gujinghui One question, will all these changes affect the performance?

**Comment by gujinghui** [comment]
@yinghai In first iteration, the buffer copy must happen.
And no changes in next iterations.

**Comment by jgong5** [comment]
> In first iteration, the buffer copy must happen. And no changes in next iterations.
@gujinghui @yinghai I think this behavior is the same as the code without the changes.

**Review by yinghai — APPROVED** [review]

**Comment by goldsborough** [comment]
You added a file `utils/thread_name.cc` to `caffe2/utils/CMakeLists.txt` which does not exist

**Comment by samson-wang** [comment]
Compared to `commit 06fa332e2b285b1ac82df9200a83c22821812ab9` version. The running time of detectron training explodes!

caffe2 on `commit 06fa332e2b285b1ac82df9200a83c22821812ab9`, the running time is about **`0.33s/iter`**. Details as following.
```
{"accuracy_cls": 0.803711, "eta": "0:43:54", "iter": 120, "loss": 1.639812, "loss_bbox": 0.526793, "loss_cls": 0.421929, "loss_mask": 0.533060, "loss_rpn_bbox_fpn2": 0.000000, "loss_rpn_bbox_fpn3": 0.005097, "loss_rpn_bbox_fpn4": 0.066229, "loss_rpn_bbox_fpn5": 0.012779, "loss_rpn_bbox_fpn6": 0.000000, "loss_rpn_cls_fpn2": 0.000370, "loss_rpn_cls_fpn3": 0.031920, "loss_rpn_cls_fpn4": 0.013575, "loss_rpn_cls_fpn5": 0.006518, "loss_rpn_cls_fpn6": 0.000000, "lr": 0.004933, "mb_qsize": 64, "mem": 5392, "time": 0.334365}
```
caffe2 on `commit 76f22b7aef9f1655f91606c3459498ec813f67f3`,  the running time is about **`0.997s/iter`**!
```
son_stats: {"accuracy_cls": 0.803711, "eta": "2:10:59", "iter": 120, "loss": 1.612624, "loss_bbox": 0.506494, "loss_cls": 0.420890, "loss_mask": 0.523235, "loss_rpn_bbox_fpn2": 0.000000, "loss_rpn_bbox_fpn3": 0.006633, "loss_rpn_bbox_fpn4": 0.066858, "loss_rpn_bbox_fpn5": 0.011601, "loss_rpn_bbox_fpn6": 0.000000, "loss_rpn_cls_fpn2": 0.000266, "loss_rpn_cls_fpn3": 0.030577, "loss_rpn_cls_fpn4": 0.017864, "loss_rpn_cls_fpn5": 0.004509, "loss_rpn_cls_fpn6": 0.000000, "lr": 0.004933, "mb_qsize": 64, "mem": 5106, "time": 0.997423}
```

**Comment by samson-wang** [comment]
@yinghai I have no idea of IDEEP engine. Same detectron code and configurations. After pull of the latest `pytorch`, the training slowed down.

**Comment by yinghai** [comment]
Thanks for reporting. Between 06fa33 and 76f22b, there are loads of commits. The regression is important to us but not necessarily related to this PR. I suggest you file an issue and state the problem. I guess given the number of commits in between. Bisection is needed to pinpoint the culprit. Thanks.

**Pull request merged**